ui/app/components/global-search/control.js

```diff
@@ -57,32 +58,40 @@ export default class GlobalSearchControl extends Component {
     const allTaskGroupResults = results.Matches.groups || [];
     const allCSIPluginResults = results.Matches.plugins || [];
 
-    const jobResults = allJobResults.slice(0, MAXIMUM_RESULTS).map(({ ID: name, Scope: [ namespace, id ]}) => ({
-      type: 'job',
-      id,
-      namespace,
-      label: name,
-    }));
-
-    const nodeResults = allNodeResults.slice(0, MAXIMUM_RESULTS).map(({ ID: name, Scope: [ id ]}) => ({
-      type: 'node',
-      id,
-      label: name,
-    }));
-
-    const allocationResults = allAllocationResults.slice(0, MAXIMUM_RESULTS).map(({ ID: name, Scope: [ , id ]}) => ({
-      type: 'allocation',
-      id,
-      label: name,
-    }));
-
-    const taskGroupResults = allTaskGroupResults.slice(0, MAXIMUM_RESULTS).map(({ ID: id, Scope: [ namespace, jobId ]}) => ({
-      type: 'task-group',
-      id,
-      namespace,
-      jobId,
-      label: id,
-    }));
+    const jobResults = allJobResults
+      .slice(0, MAXIMUM_RESULTS)
+      .map(({ ID: name, Scope: [namespace, id] }) => ({
+        type: 'job',
+        id,
+        namespace,
+        label: `${name} > ${namespace}`,
+      }));
+
+    const nodeResults = allNodeResults
+      .slice(0, MAXIMUM_RESULTS)
+      .map(({ ID: name, Scope: [id] }) => ({
+        type: 'node',
+        id,
+        label: name,
+      }));
+
+    const allocationResults = allAllocationResults
+      .slice(0, MAXIMUM_RESULTS)
+      .map(({ ID: name, Scope: [namespace, id] }) => ({
+        type: 'allocation',
+        id,
+        label: `${name} > ${namespace}`,
+      }));
+
+    const taskGroupResults = allTaskGroupResults
+      .slice(0, MAXIMUM_RESULTS)
+      .map(({ ID: id, Scope: [namespace, jobId] }) => ({
+        type: 'task-group',
+        id,
+        namespace,
+        jobId,
+        label: id,
```

Comment:
Since we now have a information hierarchy, should these results be `${namespace} > ${job} > ${id}`? Or maybe `${namespace} > ${job}.${id}` to keep it consistent with alloc results?